### DISCUSSION TITLE
A question about Heat conduction solve using displaced mesh

### DISCUSSION THREAD

**Discussion by xuxiaobei1995** [idaholab/moose#21451] [Q&A Modules: General] [answered]
Hi all,

I'm confused about an issue on the heat conduction solve using displaced mesh. Take a simple thermo-mechanical coupled case as an example: 

![case_figure](https://user-images.githubusercontent.com/34068460/176403668-0fcc6777-0d8c-487d-9935-96ca00cc6514.PNG)

It is an elongated rectangle with a uniform heat source; the geometry and condition is shown in the figure above. For this case, after the elongation, the analytical solution of the left side temperature is 2 (Tleft - Tright  = 0.5 * qv * conduction_length / k = 1). However, the result from MOOSE is 2.346. If I set use_displaced_mesh = true in the heat source kernel, the result is 3, which is still different from the analytical solution. Does any one know how these results are computed? The input file of this case is shown below, which can be run in any application including the heat conduction & tensor mechanics modules. 

```

[GlobalParams]
  displacements = 'disp_x disp_y'
[]

[Mesh]
  type = GeneratedMesh
  dim = 2
  nx = 10
  ny = 10
  xmin = 0
  xmax = 1
  ymin = 0
  ymax = 1
[]

[Variables]
  [temp]
  []
[]

[Kernels]
  [heatconduction]
    type = HeatConduction
    variable = temp
  []
  [heatsource]
    type = HeatSource
    variable = temp
    value = 1
    #use_displaced_mesh = true
  []
[]

[Modules/TensorMechanics/Master]
  [./tm]
    strain = FINITE
    add_variables = true
    decomposition_method = EigenSolution
    use_displaced_mesh = true
  [../]
[]

[Materials]
  [./elasticity_tensor]
    type = ComputeIsotropicElasticityTensor
    youngs_modulus = 2.1e5
    poissons_ratio = 0.3
  [../]
  [./stress]
    type = ComputeFiniteStrainElasticStress
  [../]
  [./hc_mat]
    type = HeatConductionMaterial
    thermal_conductivity = 1.0
  []
[]

[BCs]
  [./left]
    type = PresetBC
    variable = disp_x 
    boundary = left
    value = 0.0
  [../]
  [./bottom]
    type = PresetBC
    variable = disp_y 
    boundary = bottom
    value = 0.0
  [../]
  [./right]
    type = FunctionPresetBC
    variable = disp_x 
    boundary = right
    function = '0.02*t'
  []
  [./right_temp]
    type = PresetBC
    variable = temp
    boundary = right
    value = 1.0
  []
[]

[Preconditioning]
  [./SMP]
    type = SMP
    full = true
  [../]
[]

[Executioner]
  type = Transient
  dt = 1
  end_time = 50

  solve_type = 'PJFNK'

  petsc_options_iname = '-pc_type -sub_pc_type -pc_asm_overlap -ksp_gmres_restart'
  line_search = none
  petsc_options_value = 'asm lu 20 101'
[]

[Outputs]
  exodus = true
  csv = true
[]

```

**Comment by hugary1995** [comment, marked as answer]
```
T,xx + 1 = 0
T = -0.5x^2 + ax + b
T(2) = 1, so 2a + b = 3
T,x(0) = 0, so a = 0, b = 3
T = -0.5x^2 + 3
T(0) = 3
```
Cheers

**Reply by xuxiaobei1995** [reply]
Sorry, I just found I made a mistake when solving the analytical solution... T(0) = 3 is the right result. Big thanks to you!

**Reply by js-jixu** [reply]
I'm also studying thermomechanical coupling now, and I have some questions. Can I connect with you? My email address is on my homepage.
